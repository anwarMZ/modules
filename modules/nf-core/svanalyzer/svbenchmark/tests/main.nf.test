// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core modules test svanalyzer/svbenchmark
nextflow_process {

    name "Test Process SVANALYZER_SVBENCHMARK"
    script "../main.nf"
    process "SVANALYZER_SVBENCHMARK"

    tag "modules"
    tag "modules_nfcore"
    tag "svanalyzer"
    tag "svanalyzer/svbenchmark"

    test("homo_sapiens - illumina - vcf.gz") {

        when {
            params {
                outdir = $outputDir
            }
            process {
                """
                    input[0] = [
                        [ id:'test' ], // meta map
                        file(params.test_data['homo_sapiens']['illumina']['test_test2_paired_mutect2_calls_vcf_gz'], checkIfExists: true)
                        ]
                    input[1] = [
                        [ id:'test2' ], // meta map
                        file(params.test_data['homo_sapiens']['illumina']['test2_haplotc_vcf_gz'], checkIfExists: true)
                        ]
                    input[2] = [
                        file(params.test_data['homo_sapiens']['genome']['genome_21_fasta'], checkIfExists: true)
                        ]
                    input[3] = [
                        file(params.test_data['homo_sapiens']['genome']['genome_21_fasta_fai'], checkIfExists: true)
                        ]
                    input[4] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }
    test("homo_sapiens - illumina - vcf.gz - bed") {

        when {
            params {
                outdir = $outputDir
            }
            process {
                """
                    input[0] = [
                        [ id:'test' ], // meta map
                        file(params.test_data['homo_sapiens']['illumina']['test_test2_paired_mutect2_calls_vcf_gz'], checkIfExists: true)
                        ]
                    input[1] = [
                        [ id:'test2' ], // meta map
                        file(params.test_data['homo_sapiens']['illumina']['test2_haplotc_vcf_gz'], checkIfExists: true)
                        ]
                    input[2] = [
                        file(params.test_data['homo_sapiens']['genome']['genome_21_fasta'], checkIfExists: true)
                        ]
                    input[3] = [
                        file(params.test_data['homo_sapiens']['genome']['genome_21_fasta_fai'], checkIfExists: true)
                        ]
                    input[4] = [
                        file(params.test_data['homo_sapiens']['genome']['genome_21_multi_interval_bed'], checkIfExists: true)
                        ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
