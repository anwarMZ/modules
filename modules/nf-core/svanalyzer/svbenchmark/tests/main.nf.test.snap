{
    "distances": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.distances:md5,b6a908b1187f159429dc8c6beb6e6f57"
                ]
            ]
        ],
        "timestamp": "2023-11-27T13:23:32.512549414"
    },
    "bed_fps": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.falsepositives.vcf:md5,146f05125330a9c0c4581a5a581744a7"
                ]
            ]
        ],
        "timestamp": "2023-11-27T13:24:07.113426884"
    },
    "versions": {
        "content": [
            [
                "versions.yml:md5,6ac08491dbe591d24615c131658cf771"
            ]
        ],
        "timestamp": "2023-11-27T13:23:32.528348073"
    },
    "bed_distances": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.distances:md5,95f7b1679e805979f0c9a4a322cb77f4"
                ]
            ]
        ],
        "timestamp": "2023-11-27T13:24:07.121477323"
    },
    "bed_versions": {
        "content": [
            [
                "versions.yml:md5,6ac08491dbe591d24615c131658cf771"
            ]
        ],
        "timestamp": "2023-11-27T13:24:07.138502322"
    },
    "bed_fns": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.falsenegatives.vcf:md5,c04105833d5f9acb5ec50af0b89a6b0f"
                ]
            ]
        ],
        "timestamp": "2023-11-27T13:24:07.088409666"
    },
    "fps": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.falsepositives.vcf:md5,146f05125330a9c0c4581a5a581744a7"
                ]
            ]
        ],
        "timestamp": "2023-11-27T13:23:32.506607404"
    },
    "report": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.report:md5,d0c7db2466442cbdd4f87ad516e00d32"
                ]
            ]
        ],
        "timestamp": "2023-11-27T13:23:32.519687833"
    },
    "fns": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.falsenegatives.vcf:md5,7a574ad3c614f0c38a6dccb5901ddc05"
                ]
            ]
        ],
        "timestamp": "2023-11-27T13:23:32.493424415"
    },
    "bed_report": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.report:md5,aaa9225dd6e261951c020e23473375b9"
                ]
            ]
        ],
        "timestamp": "2023-11-27T13:24:07.130364912"
    }
}